### PR TITLE
feat: Add ref with focus to code editor

### PR DESCRIPTION
**Pull request by taheramr**
### Description

Adding a `ref` to the code editor with `focus` function to allow customers to have the ability to focus on the code editor component when needed.

Related links, issue #, if available: n/a
AWSUI-20473

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by timogasda — CHANGES_REQUESTED** [review]
Looks like there are some failing tests. I would also like us to add a new test for the new focus function.

File: src/code-editor/index.tsx
Comment (just-boris) [inline src/code-editor/index.tsx:44]: This code gives external consumers access to the entire `<div>` node and this is not what we want.

Please check how other components implement focus feature (hint, there is `useFowardFocus` hook)

**Review by just-boris — APPROVED** [review]
Thanks for addressing my comment, looks good now. 

(not sure if you need to wait for Timo approval or not)

File: src/code-editor/index.tsx
Comment (taheramr) [inline src/code-editor/index.tsx:105]: Just a small change to finalise the PR, I had to add the `props.loading` as a dependency because the codeEditorRef is not initiated when the `props.loading` is false as the div holding the codeEditorRef as a ref is conditionally rendered based on that loading value.

File: src/code-editor/__tests__/code-editor.test.tsx
Comment (timogasda) [inline src/code-editor/__tests__/code-editor.test.tsx:39]: minor typo: *programmatically

**Review by timogasda — COMMENTED** [review]
looks good apart from a tiny typo.

File: src/code-editor/index.tsx
Comment (timogasda) [reply src/code-editor/index.tsx:45]: good catch, thanks boris!

**Review by timogasda — APPROVED** [review]

**Pull request merged**
